### PR TITLE
[LIB-667] Add possibility to save user profile when saving the user.

### PR DESCRIPTION
**Pull request by opalczynski**

### CODE REVIEW

File: syncano/models/fields.py [hunk 1]
```diff
@@ -504,11 +505,15 @@ def validate(self, value, model_instance):
         super(ModelField, self).validate(value, model_instance)
 
         if not isinstance(value, (self.rel, dict)):
-            raise self.ValidationError('Value needs to be a {0} instance.'.format(self.rel.__name__))
+            if not self.is_data_object_mixin:
```
Comment (23doors) [inline syncano/models/fields.py]: Maybe `if not isinstance(value, (self.rel, dict)) and not self.is_data_object_mixin:`?

File: syncano/models/fields.py [hunk 2]
```diff
@@ -504,11 +505,15 @@ def validate(self, value, model_instance):
...
+                raise self.ValidationError('Value needs to be a {0} instance.'.format(self.rel.__name__))
 
         if self.required and isinstance(value, self.rel):
             value.validate()
 
+        if self.is_data_object_mixin and hasattr(value, 'validate'):
```
Comment (23doors) [inline syncano/models/fields.py]: Consider combining it with upper part to always call validate once
`if (self.required and isinstance(value, self.rel)) or (self.is_data_object_mixin and hasattr(value, 'validate')):`

File: syncano/models/fields.py [hunk 3]
```diff
@@ -534,6 +539,10 @@ def to_native(self, value):
             pk_value = getattr(value, pk_field.name)
             return pk_field.to_native(pk_value)
 
+        if self.is_data_object_mixin:
+            if not self.just_pk and hasattr(value, 'to_native'):
```
Comment (23doors) [inline syncano/models/fields.py]: Why nested if again? ;)

File: syncano/models/fields.py (hunk 1 above)
Comment (opalczynski) [inline syncano/models/fields.py]: :+1:

File: syncano/models/fields.py (hunk 2 above)
Comment (opalczynski) [inline syncano/models/fields.py]: :+1:

File: syncano/models/fields.py (hunk 3 above)
Comment (opalczynski) [inline syncano/models/fields.py]: :+1:

**Comment by 23doors** [comment]
seems there are some merge conflicts, but overall, looks good.

**Pull request merged**